Added pre-commit config file

Raviv-S opened the pull request:
Signed-off-by: Raviv Schaffer <raviv.schaffer@ibm.com>

Raviv-S committed:
Added pre-commit config file

Signed-off-by: Raviv Schaffer <raviv.schaffer@ibm.com>

roytman reviewed (commented):
You did not add the lint action

Raviv-S commented:
What do you mean by "adding lint action"?
As I understand it, all you need for it to work is 3 files:
1. `.pre-commit-config.yaml` which I added at this PR and as golangci-lint configured to it
2. `.git/hooks/pre-commit` which is installed locally for each user when you run `pre-commit install`
3. The hook itself, in this instance it is `.golangci.yml` which I added in previous PR.

roytman commented:
I mean to add automatic GitHub action which will run on each PR.
you can replace `.github/workflows/golang-security.yml` by your action. (gosec is a part of your lints)

Raviv-S commented:
Added that workflow on a different PR [https://github.com/fybrik/data-movement-operator/pull/53](53#).

roytman commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,12 @@
+      types: [ go ]
+      language: golang
+      pass_filenames: false
+      verbose: true
```
pls add a new line
